### ISSUE TITLE
Secret `yourls.existingSecret` does not contain `yourls-password`

### ISSUE THREAD

**Issue by jbouse** [YOURLS/charts#120]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) guide.

### Version

1.8.2

### Description

While attempting to do a Helm chart upgrade from v4.2.2 to v4.2.3 encountered errors while trying to provide existing credentials. Looking into the `yourls` secret created within the k8s cluster I noted that it only contains `password` and `username` while the use of the `existingSecret` value to point to it is expecting to find `yourls-password`

### Expectation

When attempting to do a Helm upgrade without specifying the `yourls.password` or `yourls.existingSecret` you get presented with the following:
```
    'yourls.password' must not be empty, please add '--set yourls.password=$YOURLS_PASSWORD' to the command. To get the current value:

        export YOURLS_PASSWORD=$(kubectl get secret --namespace "yourls" yourls -o jsonpath="{.data.yourls-password}" | base64 --decode)
```

However, if you look at `charts/yourls/templates/secrets.yaml`, it does not create the `yourls-password` key, so there is a disconnect in expected values. Either the `yourls-password` key needs to be created or the subsequent validation and usage checks updated to look for `password` instead`.

### Reproduction steps

1. Run `helm install yourls yourls/yourls --version 4.2.2 --namespace yourls-test --create-namespace`
2. Run `helm upgrade yourls yourls/yourls --version 4.2.3 --namespace yourls-test --set mysql.auth.existingSecret=yourls-mysql --set yourls.existingSecret=yourls`
3. Witness the failure
4. Run `kubectl get secret --namespace yourls-test yourls -o jsonpath="{.data.yourls-password}"` see it shows no value
5. Run `kubectl get secret --namespace yourls-test yourls -o jsonpath="{.data.password}"` confirm it does have a value

**Comment by LeoColomb** [comment]
Thanks for opening this issue @jbouse.
That said it is quite curious it didn't happen before, as the secret typed changed with release 4.2.0.
The new secret format is the standard `kubernetes.io/basic-auth` one:

```yaml
apiVersion: v1
kind: Secret
metadata:
  ...
type: kubernetes.io/basic-auth
data:
  username: {{ .Values.yourls.username | b64enc | quote }}
  password: {{ .Values.yourls.password | b64enc | quote }}
```

This provides more flexibility and match security standard by "secretizing" the whole account instead of just the password.

Do you confirm by updating the secret to the new format it resolves your issue?

You have a look at the diff here: https://github.com/YOURLS/charts/compare/yourls-4.1.0...yourls-4.2.0

**Comment by jbouse** [comment]
@LeoColomb Yes, the `yourls` k8s Secret does match the `kubernetes.io/basic-auth` standard format... The issue I'm seeing is that the chart still has references to the old `yourls-password` key that is causing issues with upgrading between versions. Unless I'm attempting the upgrade wrong, the steps I mentioned above are using defaults and it fails.

**Comment by LeoColomb** [comment]
Oh ok, got it, good catch indeed, thanks for that.
There is a reference in the NOTES.txt which generate the unexpected output.

https://github.com/YOURLS/charts/blob/8ce6a6b78371aa53e35141df99088d987d90ecc1/charts/yourls/templates/NOTES.txt#L95

Would you like to open a PR with a fix?

**Comment by jbouse** [comment]
When I attempt the upgrade as mentioned in step 2 of the issue I get the following:
```
helm upgrade yourls yourls/yourls --version 4.2.3 --namespace yourls-test --set mysql.auth.existingSecret=yourls-mysql --set yourls.existingSecret=yourls
Error: UPGRADE FAILED: execution error at (yourls/templates/NOTES.txt:107:4):
PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
OK
```
I get similar if instead of providing the `existingSecret`'s I instead use `--set mysql.auth.rootPassword=$MYSQL_ROOT_PASSWORD --set mysql.auth.password=$MYSQL_PASSWORD --set yourls.password=$YOURLS_PASSWORD` after running the appropriate export kubectl commands.

Thus far the only means to upgrade the chart is to uninstall and re-install with the new version. Which is quite an annoyance as it removes any existing links

**Comment by jbouse** [comment]
> Would you like to open a PR with a fix?

I can definitely throw a PR together for it as I believe it does look like a simple fix.

**Comment by jbouse** [comment]
Hmm... even with the fix upgrading the release still fails with the error about the `yourls/templates/NOTES.txt:107:4`

**Comment by LeoColomb** [comment]
Weird. I'll have a deeper look tomorrow, thanks for your report.

**Comment by jbouse** [comment]
I'm looking as well... The error refers to the final line in the NOTES.txt validating the errors so it seems like the issue is in logic between https://github.com/YOURLS/charts/blob/main/charts/yourls/templates/NOTES.txt#L92-L98.

**Comment by jbouse** [comment]
I think I've worked the solution out. Please check out #122 as to what I came up with. I did rework a lot of the end of the NOTES.txt and took out the MySQL-related validation as I found it got picked up automatically by the Bitnami MySQL chart in my testing.

I kinda get like a dog with a bone when I have something that isn't working as expected so I couldn't get to sleep until I'd figured out what was going on here.

**Comment by LeoColomb** [comment]
@jbouse I've just tried to upgrade on my k8s cluster and it seems to work fine after #123.
Do you confirm?

Not sure why the chart-testing tool refuses to validate the upgrade even if there is no error, I'll figure out in another branch (#126).

**Comment by jbouse** [comment]
@LeoColomb yes, I was able to upgrade from 4.2.2 up to 4.2.6 fine... I was trying to get the `ct install --upgrade` working as well but was having trouble is it needs to run normally without any special options for the previous version then run the upgrade of the new version with the `mysql.auth.existingSecret` and `yourls.existingSecret` but that gets generated to the name of the release which is random each time ct runs.
